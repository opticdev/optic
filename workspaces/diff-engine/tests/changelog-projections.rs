use optic_diff_engine::{SpecEvent, ChangelogProjection, ChangelogQuery};
use chrono::Utc;
use insta::assert_debug_snapshot;

#[test]
fn empty_events_gives_empty_projection() {
  let spec_events: Vec<SpecEvent> = Default::default();
  let changelog_projection = ChangelogProjection::from(spec_events);
  assert_debug_snapshot!(changelog_projection);
}

#[test]
fn can_query_empty_projection_for_added_endpoints() {
  let spec_events: Vec<SpecEvent> = Default::default();
  let changelog_projection = ChangelogProjection::from(spec_events);
  let changelog_query = ChangelogQuery{changelog: changelog_projection};
  assert_debug_snapshot!(changelog_query.added(Utc::now()));
}

#[test]
fn can_query_empty_projection_for_changed_endpoints() {
  let spec_events: Vec<SpecEvent> = Default::default();
  let changelog_projection = ChangelogProjection::from(spec_events);
  let changelog_query = ChangelogQuery{changelog: changelog_projection};
  assert_debug_snapshot!(changelog_query.changed(Utc::now()));
}