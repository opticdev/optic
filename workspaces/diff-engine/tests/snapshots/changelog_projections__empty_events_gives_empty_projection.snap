---
source: workspaces/diff-engine/tests/changelog-projections.rs
expression: changelog_projection
---
ChangelogProjection {
    endpoints: [],
}
